arangod/IResearch/IResearchLink.cpp

```diff
@@ -2017,59 +577,61 @@ AnalyzerPool::ptr IResearchLink::findAnalyzer(AnalyzerPool const& analyzer) cons
   return nullptr;
 }
 
-IResearchLink::LinkStats IResearchLink::stats() const {
-  // '_dataStore' can be asynchronously modified
-  auto lock = _asyncSelf->lock();
-  return statsUnsafe();
+std::string_view IResearchLink::format() const noexcept {
+  return getFormat(LinkVersion{_meta._version});
 }
 
-IResearchLink::LinkStats IResearchLink::statsUnsafe() const {
-  LinkStats stats;
-  if (!_dataStore) {
-    return {};
-  }
-  stats.numBufferedDocs = _dataStore._writer->buffered_docs();
-
-  // copy of 'reader' is important to hold reference to the current snapshot
-  auto reader = _dataStore._reader;
-  if (!reader) {
-    return {};
-  }
-
-  stats.numSegments = reader->size();
-  stats.numDocs = reader->docs_count();
-  stats.numLiveDocs = reader->live_docs_count();
-  stats.numFiles = 1;  // +1 for segments file
+IResearchViewStoredValues const& IResearchLink::storedValues() const noexcept {
+  return _meta._storedValues;
+}
 
-  auto visitor = [&stats](std::string const& /*name*/, irs::segment_meta const& segment) noexcept {
-    stats.indexSize += segment.size;
-    stats.numFiles += segment.files.size();
-    return true;
-  };
+const std::string& IResearchLink::getViewId() const noexcept {
+  return _viewGuid;
+}
 
-  reader->meta().meta.visit_segments(visitor);
-  return stats;
+std::string IResearchLink::getDbName() const {
+  return std::to_string(_collection.vocbase().id());
```

Comment:
Is it expected that we return the database ID here and not the database name?

Reply:
Good catch! Db name is much more user-friendly